### ISSUE TITLE
Header Regex Match: BASE64 in From and Subject Fields doesn't match

### ISSUE THREAD

**Issue by infinitelyintrigued** [RealRaven2000/FiltaQuilla#308]
When I receive a message from Microsoft Teams (noreply@email.teams.microsoft.com), I notice the sender fields contain BASE64 data like this one. (View > Message Source).
```
From: "=?utf-8?B?Q2FycmV5LEppbSAuIGluIFRlYW1zCg==?="
 <noreply@email.teams.microsoft.com>
```
The Header RegEx Match against the BASE64 data does not seem to work.
```
sender:/Jim/
```
Is there a way to do this?

Thank you

**Comment by RealRaven2000** [comment]
thanks for the report - if it is a list email would it be possible for you to export the mail as an "eml" file and post it here. If there is private info in there you can also send it to me off list.

**Comment by RealRaven2000** [comment]
Can you forward an example please? It's much easier with real life test data rather than making up stuff artificially.

**Comment by RealRaven2000** [comment]
I added some debug code so you can check what is retrieved from the subject and from the regular expression:

[filtaquilla-4.2.1pre3.zip](https://github.com/user-attachments/files/17576055/filtaquilla-4.2.1pre3.zip)

Download the zip file and drag it into Thunderbird Add-ons manager (without unpacking).

To get the result open FiltaQuilla settings on the last tab, activate Debug and the new `regexSubject` debug switch:

![image](https://github.com/user-attachments/assets/b9d2cba1-b290-41c4-a5e6-512bed783389)

After that please restart Thunderbird to ensure FQ reading the debug settings, then run your filter. You can read the error log via CTRL+SHIFT+J. I am getting this error log with the test mail you sent:

![image](https://github.com/user-attachments/assets/18fb6543-8536-406b-b174-90d5c04c86ad)

**Comment by RealRaven2000** [comment]
I think the bug may be caused by using the header regex (rather than subject regex) -  I just found out that it does not decode the sender if you do something like this:

![image](https://github.com/user-attachments/assets/d3a345bf-cd62-48f2-ac7d-92c79115e975)

I can try to use `nsIMimeConverter` to decode the "From" header.

**Comment by RealRaven2000** [comment]
Here is a test version that adds decoding for regex headers:

[filtaquilla-4.2.1pre5.zip](https://github.com/user-attachments/files/17607201/filtaquilla-4.2.1pre5.zip)

----

To try out this version, download the zip file and then drag it into Thunderbird Add-ons Manager (without extracting)

**Comment by infinitelyintrigued** [comment]
It's working! I'll continue testing it.

**Comment by infinitelyintrigued** [comment]
It's been about 5 days, but I experienced no issues. It's still working great!

**Comment by RealRaven2000** [comment]
Fixed in 4.3 - Published 17/11/2024
